filter.js

```diff
@@ -0,0 +1,14 @@
+/**
+ *
+ * @param {array|object} collection - The collection to filter.
+ * @param {function} requirement - The requirement to filter with.
+ * @returns {array} - Returns new array with elements passing the requirement provided for each iteration.
+ */
+
+export default function filter(collection, requirement) {
+  if (Array.isArray(collection)) {
+    return collection.filter((n) => requirement(n))
+  }
+  const objToArray = Object.values(collection)
+  return objToArray.filter((n) => requirement(n))
```

Comment:
Whilst I really like seeing you become more comfortable with iterables this doesn't make sense to me

`filter` is intended for arrays, not objects. I would check if not array and return empty array.

What you are doing with `Object.values` is more akin to [_.omitBy](https://lodash.com/docs/4.17.15#omitBy)

Reply:
Lodash filter is for arrays and objects:
https://lodash.com/docs/4.17.15#filter